bucket/nano.json

```diff
@@ -1,32 +1,33 @@
-{
-    "version": "7.1-12",
-    "description": "A small and friendly GNU editor, inspired by Pico",
-    "homepage": "https://www.nano-editor.org/",
-    "license": "GPL-3.0-only",
-    "notes": [
-        "Configure nano by editing its configuration file %USERPROFILE%\\.nanorc",
-        "NOTE: The config file was named nano.rc in the 2.5.3 release, but is now named .nanorc"
-    ],
-    "url": "https://files.lhmouse.com/nano-win/nano-win_10155_v7.1-12-g14f292a7c.7z",
-    "hash": "dce659505c33f6def27048d44d5998dcbe70ae4361c55fe32398af33bc83682e",
-    "post_install": [
-        "if (-not (Test-Path \"$env:USERPROFILE\\.nanorc\")) {",
-        "   Copy-Item \"$dir\\.nanorc\" \"$env:USERPROFILE\\.nanorc\"",
-        "}"
-    ],
-    "checkver": {
-        "url": "https://files.lhmouse.com/nano-win/?C=N&O=D",
-        "regex": "nano-win_(?<build>\\d+)_v([\\d.-]+)-(?<commit>\\w+)\\.7z"
-    },
-    "architecture": {
-        "64bit": {
-            "bin": "pkg_x86_64-w64-mingw32\\bin\\nano.exe"
-        },
-        "32bit": {
-            "bin": "pkg_i686-w64-mingw32\\bin\\nano.exe"
-        }
-    },
-    "autoupdate": {
-        "url": "https://files.lhmouse.com/nano-win/nano-win_$matchBuild_v$version-$matchCommit.7z"
-    }
-}
+{
+    "version": "7.2-8-10163.1",
+    "description": "The legendary small and friendly GNU editor, now UTF-8 ready for Windows 32/64",
+    "homepage": "https://www.nano-editor.org/",
+    "license": "GPL-3.0-only",
+    "notes": [
+        "Configure nano by editing its configuration file with nano ~/.nanorc",
+        "Visit https://github.com/okibcn/nano-for-windows for more information."
+    ],
+    "url": "https://github.com/okibcn/nano-for-windows/releases/download/v7.2-8-10163.1/nano-for-windows_v7.2-8-10163.1.7z",
+    "hash": "3c96a9218bda9257e20523ab37e07353104c9703e324e1198cb2c1ad308650ad",
+    "bin": "nano.exe",
+    "pre_install": [
+        "$arch=if ($architecture -match 64) {'x86_64'} else {'i686'}",
+        "$scoopdir=$scoopdir -replace '\\\\', '/'",
+        "if (-not (Test-Path ~/.nanorc)) {",
+        "    Move-Item \"$dir/.nanorc\" ~/.nanorc",
+        "    Add-Content ~/.nanorc \"include `\"$scoopdir/apps/nano/current/syntax/*.nanorc`\"\"}",
+        "Move-Item \"$dir/pkg_$arch-w64-mingw32/bin/*\" \"$dir\"",
+        "Move-Item \"$dir/pkg_$arch-w64-mingw32/share/nano\" \"$dir/syntax\" -Force",
+        "Remove-Item  \"$dir/pkg*\" -Force -Recurse",
+        "(Get-Content ~/.nanorc) -replace '(@PKGDATADIR@)', \"$scoopdir/apps/nano/current/syntax\" | Set-Content ~/.nanorc",
+        "(Get-Content ~/.nanorc) -replace '(pkg_.*-w64-mingw32.share.nano)', 'syntax' | Set-Content ~/.nanorc",
+        "(Get-Content ~/.nanorc) -replace '(/usr/share/nano)', \"$scoopdir/apps/nano/current/syntax\" | Set-Content ~/.nanorc"

```

Comment:
Do not modify user's existing rcfile
``` suggestion
        "$prefix = $scoopdir -replace '\\\\', '/'",
        "if (-not (Test-Path ~/.nanorc)) {",
        "    Add-Content \"$dir/.nanorc\" \"include `\"$prefix/apps/nano/current/syntax/*.nanorc`\"\"",
        "    (Get-Content \"$dir/.nanorc\") -replace '(@PKGDATADIR@)', \"$prefix/apps/nano/current/syntax\" | Set-Content ~/.nanorc",
        "    (Get-Content \"$dir/.nanorc\") -replace '(pkg_.*-w64-mingw32.share.nano)', 'syntax' | Set-Content ~/.nanorc",
        "    (Get-Content \"$dir/.nanorc\") -replace '(/usr/share/nano)', \"$prefix/apps/nano/current/syntax\" | Set-Content ~/.nanorc",
        "    Move-Item \"$dir/.nanorc\" ~/.nanorc",
        "}",
        "Move-Item \"$dir/pkg_$arch-w64-mingw32/bin/*\" \"$dir\"",
        "Move-Item \"$dir/pkg_$arch-w64-mingw32/share/nano\" \"$dir/syntax\" -Force",
        "Remove-Item  \"$dir/pkg*\" -Force -Recurse"
```

Reply:
> I know it requires the underlying host to support unicode, the problem is even if the terminal (ConHost) does not support unicode n4w should not print the garbled `�[?2004h�[?2004l` text every time I `Ctrl X` quit the editor. In the gif, the last command is nano-win and it does not have such a problem. I do not know if you can reproduce but the step is simply to use WindowsPowerShell, whose backend is ConHost, to open your nano build and quit.

Thanks [chawyehsu](https://github.com/chawyehsu).  No need to track an issue, the garbage issue in non-UTF-8 ready terminals is already fixed in the new release. You can DL it from https://github.com/okibcn/nano-for-windows/releases/tag/v7.2-8-10163.2

I could reproduce the terminal state by opening poweshell.exe from 🪟+R. Thanks

Reply:
Well, I have some concerns about that suggestion. You are basically concerned about these three lines:
```pwsh
(Get-Content ~/.nanorc) -replace '(@PKGDATADIR@)', "$scoopdir/apps/nano/current/syntax" | Set-Content ~/.nanor
(Get-Content ~/.nanorc) -replace '(/usr/share/nano)', "$scoopdir/apps/nano/current/syntax" | Set-Content ~/.nanorc
(Get-Content ~/.nanorc) -replace '(pkg_.*-w64-mingw32.share.nano)', 'syntax' | Set-Content ~/.nanor
```
1. The first one fixes the problem with the different ways packages are installed in Linux and Windows. It is not a problem for a previous `.rcfile` as it is an inactive file. I don't mind not applying the fix to a previous config file, but it is a coherent fix recommended for any config file. Much better to apply it than not.
2. It is almost the same as the previous line, replacing the Linux path with the Windows path. In the same sense, it is a recommended change to avoid faulty lines in any config file, and this one fixes several faulty `.nanorc`  lines.
3. Maybe this one is the most important. It has to be applied in any case as it replaces a non-existent path in the current `.nanorc` file with the working one. It is extremely important to do so since otherwise, the update will end in a loss of functionality.

I understand your concern and I agree that there is a safer way. I suggest retaining a .nanorc.bak file with the previous configuration with a message about the update and the presence of the bak file in the notes section. The preinstall script would be this:
```pwsh
$arch=if ($architecture -match '64') {'x86_64'} else {'i686'}
$scoopdir=$scoopdir -replace '\\', '/'
if (-not (Test-Path ~/.nanorc)){
    Move-Item "$dir/.nanorc" ~/.nanorc
    Add-Content ~/.nanorc "include `"$scoopdir/apps/nano/current/syntax/*.nanorc`""}
elseif (-not (Test-Path ~/.nanorc.bak)){Move-Item ~/.nanorc ~/.nanorc.bak}
Move-Item "$dir/pkg_$arch-w64-mingw32/bin/*" "$dir\"
Move-Item "$dir/pkg_$arch-w64-mingw32/share/nano" "$dir/syntax" -Force
Remove-Item  "$dir/pkg*" -Force -Recurs
(Get-Content ~/.nanorc) -replace '(@PKGDATADIR@)', "$scoopdir/apps/nano/current/syntax" | Set-Content ~/.nanor
(Get-Content ~/.nanorc) -replace '(/usr/share/nano)', "$scoopdir/apps/nano/current/syntax" | Set-Content ~/.nanorc
(Get-Content ~/.nanorc) -replace '(pkg_.*-w64-mingw32.share.nano)', 'syntax' | Set-Content ~/.nanor
```

And the notes would be something like this:
```
Configure nano by editing its configuration file with nano ~/.nanorc
If you are upgrading nano, a copy of the old config file is available at ~/.nanorc.bak
Visit https://github.com/okibcn/nano-for-windows for more information.
```